### PR TITLE
Conversation viewer

### PR DESCRIPTION
**Pull request by PCSwingle**
TODO:

Increase aesthetic appeal
Should we make conversation command hidden? It probably isn't too useful to the average viewer, and it looks very hacked together

### CODE REVIEW

**Comment by PCSwingle** [comment]
Current state screenshot:

image

**Comment by PCSwingle** [comment]
Just uploaded a transcript with a fairly large conversation; feel free to use the transcript viewer to check it out!

[transcript_20231013_144235.log](https://github.com/AbanteAI/mentat/files/12899698/transcript_20231013_144235.log)

File: mentat/conversation.py [hunk 1]
```diff
@@ -35,8 +44,12 @@ def __init__(self):
         self.model = config.model()
         self.messages = list[dict[str, str]]()
 
+        # This contain the messages the user actually sends and the messages the model output
+        # along with a snapshot of exactly what the modelg got before that message
```
Comment (jakethekoenig) [inline mentat/conversation.py:48]: s/modelg/model/g

File: mentat/conversation.py [hunk 2]
```diff
@@ -18,13 +23,17 @@
...
 )
-from .parsers.file_edit import FileEdit
-from .parsers.parser import PARSER, Parser
-from .session_stream import SESSION_STREAM, SessionStream
+from .session_stream import SESSION_STREAM
 
 CONVERSATION: ContextVar[Conversation] = ContextVar("mentat:conversation")
 
 
+class MessageRole(Enum):
```
Comment (jakethekoenig) [inline mentat/conversation.py:31]: I think it's better to call this `Role`. Thanks for refactoring it in this way. I think it's a bit cleaner.

**Review by jakethekoenig — APPROVED** [review]
When you say hide `/conversation` you mean remove it from the autocompleter? I think that'd be a good idea. I also like the idea of using jinja for our templates. I think it's good enough to merge though. One feature requests going forward: It'd be really nice if `transcript_viewer` could automatically open the most recent transcript and if you could scroll through them with the arrow keys.

**Comment by PCSwingle** [comment]
> When you say hide `/conversation` you mean remove it from the autocompleter? I think that'd be a good idea. I also like the idea of using jinja for our templates. I think it's good enough to merge though. One feature requests going forward: It'd be really nice if `transcript_viewer` could automatically open the most recent transcript and if you could scroll through them with the arrow keys.

Oohh, I really like that idea of being able to scroll through them. I'll add an issue for both jinja and the transcript scrolling

**Pull request merged**